src/audio/pipeline/pipeline-stream.c

```diff
@@ -280,6 +282,25 @@ int pipeline_trigger(struct pipeline *p, struct comp_dev *host, int cmd)
 	switch (cmd) {
 	case COMP_TRIGGER_PAUSE:
 	case COMP_TRIGGER_STOP:
+#if CONFIG_LIBRARY
+		ret = pipeline_trigger_run(p, host, cmd);
+		if (ret >= 0) {
+			switch (cmd) {
+			case COMP_TRIGGER_PRE_START:
+			case COMP_TRIGGER_START:
+			case COMP_TRIGGER_PRE_RELEASE:
+				p->status = COMP_STATE_ACTIVE;
+				break;
+			case COMP_TRIGGER_STOP:
+			case COMP_TRIGGER_PAUSE:
+				p->status = COMP_STATE_PAUSED;
+				break;
+			default:
+				break;
+			}
+		}
+		return ret < 0 ? ret : 0;
+#endif
```

Comment:
This is not very nice when we have to add LIBRARY specific code to already complex functions, but it seems we get away with modifications to very limited number of places in the pipeline code, so I'm good with this.